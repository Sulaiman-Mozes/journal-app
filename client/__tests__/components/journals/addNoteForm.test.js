import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import AddNoteForm from '../../../components/journals/form';

describe('Provider and AddNoteForm', () => {
  const props = {
    fields: { title: '', content: '' },
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
  };

  const NoteForm = shallow(<AddNoteForm {...props} />);

  it('renders <Provider/> correctly', () => {
    expect(NoteForm).toMatchSnapshot();
  });
});
