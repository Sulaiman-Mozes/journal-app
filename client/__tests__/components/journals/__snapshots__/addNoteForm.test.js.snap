// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provider and AddNoteForm renders <Provider/> correctly 1`] = `
<section
  className="padding-top mb-4 container"
>
  <h4
    className="h4-responsive font-weight-bold text-center my-4"
  >
    Add Note
  </h4>
  <div
    className="col-md-12 mb-5"
  >
    <form
      onSubmit={[MockFunction]}
    >
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <div
            className="md-form mb-0"
          >
            <input
              className="form-control"
              id="title"
              name="title"
              onChange={[MockFunction]}
              placeholder="Title"
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <div
            className="md-form"
          >
            <textarea
              className="form-control md-textarea"
              id="content"
              name="content"
              onChange={[MockFunction]}
              placeholder="Content"
              required={true}
              rows="2"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="text-center text-md-left"
      >
        <button
          className="btn btn-primary"
          type="submit"
        >
          Add
        </button>
      </div>
    </form>
  </div>
</section>
`;
